### PR TITLE
8334653: ISO 4217 Amendment 177 Update

### PR DESCRIPTION
**Pull request by justin-curtis-lu**
Please review this PR which incorporates the ISO 4217 Amendment 177 Update.

Specifically, the introduction of the new currency, Zimbabwe Gold.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8334653](https://bugs.openjdk.org/browse/JDK-8334653): ISO 4217 Amendment 177 Update (**Bug** - P3)

### Reviewers
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19813/head:pull/19813` \
`$ git checkout pull/19813`

Update a local copy of the PR: \
`$ git checkout pull/19813` \
`$ git pull https://git.openjdk.org/jdk.git pull/19813/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19813`

View PR using the GUI difftool: \
`$ git pr show -t 19813`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19813.diff">https://git.openjdk.org/jdk/pull/19813.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19813#issuecomment-2181254126)

### CODE REVIEW

File: src/java.base/share/data/currency/CurrencyData.properties [hunk 1]
```diff
@@ -582,7 +582,7 @@ YE=YER
 # ZAMBIA
 ZM=ZMW
 # ZIMBABWE
-ZW=ZWL
+ZW=ZWL;2024-08-31-22-00-00;ZWG
```
Comment (naotoj) [inline src/java.base/share/data/currency/CurrencyData.properties:585]: JDK switches the currency at the beginning of the transition period. So in this case, it's 6/25. So we don't need to even specify the transition at all, as the next build drop is beyond that date.

File: test/jdk/java/util/Currency/tablea1.txt [hunk 2]
```diff
@@ -276,7 +276,7 @@ WF	XPF	953	0
 EH	MAD	504	2
 YE	YER	886	2
 ZM	ZMW	967	2
-ZW	ZWL	932	2
+ZW	ZWL	932	2	2024-08-31-22-00-00	ZWG	924	2
```
Comment (naotoj) [inline test/jdk/java/util/Currency/tablea1.txt:279]: Same as above

File: src/java.base/share/classes/sun/util/resources/CurrencyNames.properties [hunk 3]
```diff
@@ -512,5 +513,6 @@ yum=Yugoslavian New Dinar (1994-2002)
 zar=South African Rand
 zmk=Zambian Kwacha
 zwd=Zimbabwean Dollar (1980-2008)
+zwg=Zimbabwe Gold
```
Comment (naotoj) [inline src/java.base/share/classes/sun/util/resources/CurrencyNames.properties:516]: It's interesting to see the difference `Zimbabwe` and `Zimbabwean` here. The latter is from CLDR and the name for ZWG is not yet available, so we may need to pull the name from CLDR later.

**Comment by naotoj** [comment]
It's not about this update, but it may be better to rename the `tablea1.txt` data file to `list-one.txt` or something along the line. The name `tablea1` is obsolete, and it is now called "List one: Currency, fund and precious metal codes"
https://www.six-group.com/en/products-services/financial-information/data-standards.html

File: src/java.base/share/data/currency/CurrencyData.properties (hunk 1 above)
Comment (justin-curtis-lu) [reply src/java.base/share/data/currency/CurrencyData.properties]: Good to know. Removed the cutover dates.

File: src/java.base/share/classes/sun/util/resources/CurrencyNames.properties (hunk 3 above)
Comment (justin-curtis-lu) [reply src/java.base/share/classes/sun/util/resources/CurrencyNames.properties:516]: I see your point. CLDR takes precedence over ISO4217 regarding currencies here?

**Comment by justin-curtis-lu** [comment]
> It's not about this update, but it may be better to rename the `tablea1.txt` data file to `list-one.txt` or something along the line. The name `tablea1` is obsolete, and it is now called "List one: Currency, fund and precious metal codes" https://www.six-group.com/en/products-services/financial-information/data-standards.html

Without further context, it is hard to infer meaning from `tablea1.txt` or `list-one.txt`. I have renamed it to `ISO4217-codes.txt`, which I think is much more straight forward and provides immediate info.

File: src/java.base/share/classes/sun/util/resources/CurrencyNames.properties (hunk 3 above)
Comment (naotoj) [reply src/java.base/share/classes/sun/util/resources/CurrencyNames.properties:516]: That is more consistent IMO

**Comment by naotoj** [comment]
> > It's not about this update, but it may be better to rename the `tablea1.txt` data file to `list-one.txt` or something along the line. The name `tablea1` is obsolete, and it is now called "List one: Currency, fund and precious metal codes" https://www.six-group.com/en/products-services/financial-information/data-standards.html
> 
> Without further context, it is hard to infer meaning from `tablea1.txt` or `list-one.txt`. I have renamed it to `ISO4217-codes.txt`, which I think is much more straight forward and provides immediate info.

The point I wanted to make is that the text file only contains the "list one" part of the ISO 4217. Looks like "list two" is a subset of list one (as they are funds codes), but "list three", which lists historic codes, is definitely not covered in that file.

I am ok with the suggested name, but then we should include some comments mentioning this.

**Comment by justin-curtis-lu** [comment]
> The point I wanted to make is that the text file only contains the "list one" part of the ISO 4217. 

Renamed the file name to "ISO4217-list-one.txt"

File: src/java.base/share/classes/sun/util/resources/CurrencyNames.properties (hunk 3 above)
Comment (srl295) [reply src/java.base/share/classes/sun/util/resources/CurrencyNames.properties:516]: CLDR is aware of this change but hasn't added it yet. We're mid cycle so it won't be much more than metadata this round due to timing.  https://unicode-org.atlassian.net/issues/CLDR-17751 if you want to track.

Comment (naotoj) [reply src/java.base/share/classes/sun/util/resources/CurrencyNames.properties:516]: > CLDR is aware of this change but hasn't added it yet. We're mid cycle so it won't be much more than metadata this round due to timing. https://unicode-org.atlassian.net/issues/CLDR-17751 if you want to track.

Out of curiosity, what's the rationale behind the currency name change from "Zimbabwe Dollar" (ISO 4217 definition) to "Zimbabwean Dollar"?

**Review by naotoj — APPROVED** [review]

File: src/java.base/share/classes/sun/util/resources/CurrencyNames.properties (hunk 3 above)
Comment (srl295) [reply src/java.base/share/classes/sun/util/resources/CurrencyNames.properties:516]: > Out of curiosity, what's the rationale behind the currency name change from "Zimbabwe Dollar" (ISO 4217 definition) to "Zimbabwean Dollar"?

I'm not sure.

**Pull request closed without merging**